source/agora/consensus/EnrollmentManager.d

```diff
@@ -532,6 +532,78 @@ public class EnrollmentManager
         return false;
     }
 
+    /***************************************************************************
+
+        Get all the current validators
+
+        Params:
+            validators = will be filled with all the validators during
+                their validation cycles
+
+        Returns:
+            Return true if there is no error in getting validators
+
+    ***************************************************************************/
+
+    public bool getValidators (out Enrollment[] validators) @safe nothrow
+    {
+        try
+        {
+            () @trusted {
+                auto results = this.db.execute("SELECT val FROM validator_set" ~
+                    " WHERE enrolled_height is not null");
+                foreach (row; results)
+                {
+                    validators ~=
+                        deserializeFull!Enrollment(row.peek!(ubyte[])(0));
+                }
+            }();
+        }
+        catch (Exception ex)
+        {
+            log.error("Database operation error, exception:{}", ex);
+            return false;
+        }
+
+        return true;
+    }
+
+    /***************************************************************************
+
+        Set current block height
+
+        The enrollment manager can weed out expired validators from the set
+        based on the block height. Validators are deleted if their enrolled
+        height is less than or equal to the value of the passed block height
+        minus the validator cycle.
+
+        Params:
+            block_height = current block height
+
+    ***************************************************************************/
+
+    public void setCurrentBlockHeight (ulong block_height) @safe nothrow
+    {
+        // the smallest enrolled height would be 1, so the passed block height
+        // should be greater than the validator cycle for deleting validators.
+        if (block_height > ValidatorCycle)
```

Comment:
A style preference: Try to always return early. If you have an `if` that covers most of the function, it's a red flag.
Instead one can write:
```D
if (block_heigth <= ValidatorCycle)
    return;
```

This way you get the error and odd cases out of the way, and it makes it easier for the reader to focus on your code.
But since it's just a style preference, let's not block the merge on this.